### PR TITLE
Update tags handling in expr functions to match Graphite web

### PR DESCRIPTION
**Pull request by carrieedwards**
This PR updates expr functions to add function-specific tags to match Graphite web functions.

### CODE REVIEW

**Comment by npazosmendez** [comment]
Also, it looks like metrictank is making sure not to modify the input series: https://github.com/grafana/metrictank/blob/master/expr/func_absolute.go#L33 . Not sure how relevant that is either.

File: expr/functions/multiplySeriesWithWildcards/function.go [hunk 1]
```diff
@@ -71,6 +71,15 @@ func (f *multiplySeriesWithWildcards) Do(ctx context.Context, e parser.Expr, fro
 		groups[node] = append(groups[node], a)
 	}
 
+	commonTags := helper.CopyTags(args[0])
+	for _, serie := range args {
+		for k, v := range serie.Tags {
+			if commonTags[k] != v {
```
Comment (ywwg) [inline expr/functions/multiplySeriesWithWildcards/function.go:77]: do you need to test that k is in commonTags? or does that just return nil in Go? I forget

File: expr/functions/multiplySeriesWithWildcards/function.go [hunk 2]
```diff
@@ -107,6 +116,12 @@ func (f *multiplySeriesWithWildcards) Do(ctx context.Context, e parser.Expr, fro
 			}
 		}
 
+		if _, ok := commonTags["name"]; !ok {
```
Comment (ywwg) [inline expr/functions/multiplySeriesWithWildcards/function.go:119]: this is what I thought you might have to do above

**Review by ywwg — CHANGES_REQUESTED** [review]
looks mostly fine, just a few questions

**Comment by npazosmendez** [comment]
> Also, it looks like metrictank is making sure not to modify the input series: https://github.com/grafana/metrictank/blob/master/expr/func_absolute.go#L33 . Not sure how relevant that is either.

BUMP. This is why tests now have to use `TestEvalExpr` instead of `TestEvalExprModifiedOrigin`. Do we care if these functions change the input data?

**Comment by npazosmendez** [comment]
FTR: I'll take care of the input data immutability issue, just talked with Carrie.

 > Do we want the changes to modifying input data to be a separate PR?
I'd prefer to keep everything in this same PR,it shouldn't make the diff much larger
